app/controllers/generalshoppinglist_controller.rb

```diff
@@ -0,0 +1,38 @@
+class GeneralshoppinglistController < ApplicationController
+  def index
+    @all_recipes = Recipe.all
+    @recipes = current_user.recipes.includes(:foods)
+    @general_foods = current_user.foods
+
+    @missing_foods = calculate_missing_foods
+    @total_count = @missing_foods.length
+    @total_price = calculate_total_price(@missing_foods)
+  end
+
+  private
+
+  def calculate_missing_foods
+    missing_foods = []
+
+    @all_recipes.each do |recipe|
+      recipe.recipe_foods.each do |recipe_food|
+        food = recipe_food.food
+        general_food = @general_foods.find_by(name: food.name)
+
+        next unless general_food.nil? || general_food.quantity < recipe_food.quantity
+
+        missing_foods << {
+          name: food.name,
+          quantity: recipe_food.quantity - (general_food&.quantity || 0),
+          price: food.price
+        }
+      end
+    end
+
+    missing_foods
+  end
+
+  def calculate_total_price(foods)
+    foods.sum { |food| food[:price] * food[:quantity] }
+  end
```

Comment:
- [ ] As per the [skinny controller rule](https://dev.to/kputra/rails-skinny-controller-skinny-model-5f2k), please keep the necessary logic only in your controllers to make a request. I recommend you to move these functions into your model or helpers and so you can test them without making a request.